File: app/models/esi/download_markets_prices.rb
require_relative 'download'

# Seems not to require full download of all prices history
# Just jita, see ...
class Esi::DownloadMarketsPrices < Esi::Download

  def initialize( debug_request: false )
    super( 'markets/prices/', {}, debug_request: debug_request )
  end

  def download
    Banner.p 'About to download cpp markets prices'

    File.open( 'data/cpp_market_prices.yaml', 'w' ) do |f|
      f.puts( Hash[ get_all_pages.map{ |e| [ e['type_id'], e ] } ].to_yaml )
    end
  end

end

File: esi/download_markets_prices.py
import logging
import yaml
import os
from esi.client import EsiClient

logger = logging.getLogger(__name__)


class DownloadMarketsPrices:
    def download(self):
        client = EsiClient('markets/prices/')
        prices = client.get_all_pages()

        price_map = {}
        for p in prices:
            price_map[p['type_id']] = {
                'adjusted_price': p.get('adjusted_price'),
                'average_price': p.get('average_price'),
            }

        os.makedirs('data', exist_ok=True)
        with open('data/cpp_market_prices.yaml', 'w') as f:
            yaml.dump(price_map, f)

        logger.debug('Downloaded %s market prices', len(price_map))
